src/Log/StrftimeToICUFormatMap.php

```diff
@@ -0,0 +1,115 @@
+ * equivalents.
+ *
+ * @internal
+ *
+ * @see https://www.php.net/strftime for PHP strftime format strings
+ * @see https://unicode-org.github.io/icu/userguide/format_parse/datetime/ for ICU Date/Time format strings
+ */
+final class StrftimeToICUFormatMap
+{
+    public static function mapStrftimeToICU(string $format): string
+    {
+        return preg_replace_callback(
+            '/(?P<token>%[aAbBcCdDeFgGhHIjklmMpPrRsSTuUVwWxXyYzZ])/',
+            function (array $matches) {
+                Assert::keyExists($matches, 'token');
+                Assert::string($matches['token']);
+                switch (true) {
+                    case $matches['token'] === '%a':
+                        return 'eee';
+                    case $matches['token'] === '%A':
+                        return 'eeee';
+                    case $matches['token'] === '%b':
+                        return 'MMM';
+                    case $matches['token'] === '%B':
+                        return 'MMMM';
+                    case $matches['token'] === '%C':
+                        return 'yy';
+                    case $matches['token'] === '%d':
+                        return 'dd';
+                    case $matches['token'] === '%D':
+                        return 'MM/dd/yy';
+                    case $matches['token'] === '%e':
+                        return ' d';
+                    case $matches['token'] === '%F':
+                        return 'y-MM-dd';
+                    case $matches['token'] === '%g':
+                        return 'yy';
+                    case $matches['token'] === '%G':
+                        return 'y';
+                    case $matches['token'] === '%h':
+                        return 'MMM';
+                    case $matches['token'] === '%H':
+                        return 'HH';
+                    case $matches['token'] === '%I':
+                        return 'KK';
+                    case $matches['token'] === '%j':
+                        return 'D';
+                    case $matches['token'] === '%k':
+                        return ' H';
+                    case $matches['token'] === '%l':
+                        return ' h';
+                    case $matches['token'] === '%m':
+                        return 'MM';
+                    case $matches['token'] === '%M':
+                        return 'mm';
+                    case $matches['token'] === '%p':
+                        return 'a';
+                    case $matches['token'] === '%P':
+                        return 'a';
+                    case $matches['token'] === '%r':
+                        return ' h:mm:ss a';
+                    case $matches['token'] === '%R':
+                        return 'HH:mm';
+                    case $matches['token'] === '%S':
+                        return 'ss';
+                    case $matches['token'] === '%s':
+                        return (string) time();
```

Comment:
As I mentioned in the review, this specific call seems totally broken: we should at least use a `DateTimeImmutable` given as parameter (can re-use the instance we create in the call-site)

Reply:
Argghhh - missed that one. Have a good idea on how to fix it, too.

Reply:
Fixed now. The method now accepts a `DateTimeInterface $requestTime`, and the timestamp is retrieved from there. The value is created in `AccessLogDataMap` from the request timestamp.

Reply:
Decent trade-off: let's do it.